src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1214,29 +1219,24 @@ LedgerManagerImpl::applyTransactions(
     {
         ZoneNamedN(txZone, "applyTransaction", true);
         auto txTime = mTransactionApply.TimeScope();
-        TransactionMeta tm(2);
+        TransactionMetaFrame tm(ltx.loadHeader().current().ledgerVersion);
         CLOG_DEBUG(Tx, " tx#{} = {} ops={} txseq={} (@ {})", index,
                    hexAbbrev(tx->getContentsHash()), tx->getNumOperations(),
                    tx->getSeqNum(),
                    mApp.getConfig().toShortString(tx->getSourceID()));
         tx->apply(mApp, ltx, tm);
-
         TransactionResultPair results;
         results.transactionHash = tx->getContentsHash();
         results.result = tx->getResult();
 
-        // First gather the TransactionResultPair into the TxResultSet for
-        // hashing into the ledger header.
-        txResultSet.results.emplace_back(results);
+        txResultSetFrame.pushMetaAndResultPair(tm.getXDR(), results);
 
         // Then potentially add that TRP and its associated TransactionMeta
         // into the associated slot of any LedgerCloseMeta we're collecting.
         if (ledgerCloseMeta)
         {
-            TransactionResultMeta& trm =
-                ledgerCloseMeta->txProcessing().at(index);
-            trm.txApplyProcessing = tm;
-            trm.result = results;
+            ledgerCloseMeta->setLastTxProcessingMetaAndResultPair(tm.getXDR(),
```

Comment:
this breaks meta:
you replaced an update at index `index` by an update of the last element

Reply:
`index` always points to the last element pushed onto the meta (it starts at 0 and is ++'ed by one on each iteration of the loop, as we process transactions). There's no difference.

Reply:
when we charge fees earlier, we insert elements, so the last element is definitely not the same one

Reply:
Ah, I see, the fee-processing loop is a first pass over all the txs -- that extends the result vector -- and then the tx-applying loop does a second pass over all of those vector elements. Ok.

Reply:
(I believe this conversation is resolved by your added commits below)